app/models/feed.rb

```diff
@@ -10,7 +10,13 @@ def get(limit, max_id = nil, since_id = nil)
     if redis.exists("account:#{@account.id}:regeneration")
       from_database(limit, max_id, since_id)
     else
-      from_redis(limit, max_id, since_id)
+      statuses = from_redis(limit, max_id, since_id)
+
+      statuses += from_database(limit - statuses.size,
```

Comment:
If `limit - statuses.size` is 0, it is better not to execute `from_database`

Reply:
There is no need to perform a LIMIT 0 query. It may not be slow in postgres itself, but sending the query to postgres (and taking up a connection from the pool with e.g. pgBouncer) is still a cost that can be avoided.